rotkehlchen/tests/unit/test_trades.py

```diff
@@ -137,14 +138,61 @@ def test_query_trade_history_online_but_exchange_excluded(events_historian):
     that exchange is not connected, but is also at the excluded exchanges we
     don't end up querying trades of all exchanges"""
 
-    patch_latest_trades = patch.object(events_historian, 'query_location_latest_trades')
-    patch_iterate_exchanges = patch.object(events_historian.exchange_manager, 'iterate_exchanges')
+    exchange = create_test_binance(
+        name='Binance',
+        database=events_historian.db,
+        msg_aggregator=events_historian.msg_aggregator,
+    )
 
-    with patch_latest_trades as latest_trades_mock, patch_iterate_exchanges as iterate_exchanges_mock:  # noqa: E501
+    with (
+        patch.object(
+            target=events_historian.exchange_manager,
+            attribute='iterate_exchanges',
+        ) as iterate_exchanges_mock,
+        patch.object(
+            target=events_historian.exchange_manager,
+            attribute='connected_exchanges',
```

Comment:
This test has a long story already. Update the docstring and adjust to the changes I mentioned. This test is a regression test for https://github.com/rotki/rotki/commit/a448289f516840293cd7e82fd8efdbb3088fe1ac

```python
@pytest.mark.parametrize('db_settings', [
    {'non_syncing_exchanges': [ExchangeLocationID(name='Binance', location=Location.BINANCE)]}])
def test_query_trade_history_online_but_exchange_excluded(events_historian, function_scope_binance):
    """
    Test that if an online refresh of trades for an exchange is requested but is also at
    the excluded exchanges we don't end up querying trades of all exchanges.
    """
    with (
        patch.object(target=events_historian.exchange_manager, attribute='iterate_exchanges') as iterate_exchanges_mock,  # noqa: E501
        patch.object(target=function_scope_binance,attribute='query_trade_history') as patch_query_trade_history,  # noqa: E501
    ):
        events_historian.query_trades(
            filter_query=TradesFilterQuery.make(location=Location.BINANCE),
            only_cache=False,
        )

    assert patch_query_trade_history.call_count == 0
    assert iterate_exchanges_mock.call_count == 0
```